js/api.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable no-unused-vars */
+import { renderSimilarOffers } from './generate-elements.js';
+import {mainMarker} from './map.js';
+
+const form = document.querySelector ('.ad-form');
+const allSelectForms = form.querySelectorAll ('select');
+const allInputForms = form.querySelectorAll ('input');
+const addressInput = form.querySelector('#address');
+const filtersArea = document.querySelector('.map__filters');
+const selectFilters = filtersArea.querySelectorAll('select');
+const checkboxFilters = filtersArea.querySelectorAll('input[type="checkbox"]');
+const resetButton = document.querySelector('.ad-form__reset');
+const successMessageTemplate = document.querySelector('#success');
+const successMessage = successMessageTemplate.cloneNode(true);
+const errorMessageTemplate = document.querySelector('#error');
+const closeErrorButton = document.querySelector('.error__button');
+const errorMessage = errorMessageTemplate.cloneNode(true);
+
+const getData = (onSuccess, onFail) => {
+  fetch ('https://24.javascript.pages.academy/keksobooking/data')
+    .then((response)  => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        throw new Error('Ошибка загрузки данных');
+      }
+    })
+    .then((offers) => {
+      onSuccess (offers);
+    })
+    .catch ((err) => {
+      onFail(err);
+    });
```

Comment:
добавь лог ошибок, чтобы проще определить где искать
```suggestion
    .catch ((err) => {
      console.error(err);
      onFail(err);
    });
```